src/boost/cpp/server/device_impl.cpp

```diff
@@ -484,6 +484,16 @@ namespace PyDeviceImpl
         SAFE_PUSH_EVENT_DATE_QUALITY_VARGS(self, name, filt_names, filt_vals, data, t, quality, x, y)
     }
 
+    /* **********************************
+     * data ready event
+     * **********************************/
+    inline void push_data_ready_event(Tango::DeviceImpl &self, const str &name,
+                                      long ctr)
+    {
+        SAFE_PUSH(self, attr, name)
+        self.push_data_ready_event(__att_name, ctr); //__att_name from SAFE_PUSH
```

Comment:
I noticed that `push_archive_event` uses `attr.fire_archive_event()`. Is there any reason why you're not doing something similar, e.g `attr.fire_data_ready_event()`?